src/dotnet/commands/dotnet-cache/CacheCommand.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Cache
+{
+    public partial class CacheCommand
+    {
+        public string ProjectPath { get; set; }
+        public string Framework { get; set; }
+        public string Runtime { get; set; }
+        public string OutputPath { get; set; }
+        public string FrameworkVersion { get; set; }
+        public string IntermediateDir { get; set; }
+        public string Verbosity { get; set; }
+        private bool SkipOptimization { get; set; }
+        private bool PreserveIntermediateDir { get; set; }
+
+        public List<string> ExtraMSBuildArguments { get; set; }
+
+        private CacheCommand()
+        {
+        }
+
+        private void PrepareForExecute(CommandOption argument)
+        {
+            string targetProjectPath = string.Empty;
+
+            if (argument == null || !argument.Values.Any())
+            {
+                throw new InvalidOperationException(LocalizableStrings.SpecifyEntries);
+            }
+            try
+            {
+                if (string.IsNullOrEmpty(IntermediateDir))
+                {
+                    var tempDir = Path.GetTempPath();
+                    var folder = Path.GetRandomFileName();
+                    IntermediateDir = Path.Combine(tempDir, folder);
+                }
+                else
+                {
+                    IntermediateDir = Path.GetFullPath(IntermediateDir);
+                }
+
+                if (Directory.Exists(IntermediateDir))
```

Comment:
I see. Makes sense. In that case, should this piece of code be moved into the SDK? Otherwise, when invoking this target directly, the user will have to first create the folder and then invoke this command.